source/extensions/filters/http/grpc_web/grpc_web_filter.cc

```diff
@@ -208,6 +208,8 @@ Http::FilterTrailersStatus GrpcWebFilter::encodeTrailers(Http::HeaderMap& traile
         return Http::HeaderMap::Iterate::Continue;
       },
       &temp);
+  // Clear out the trailers so they don't get added since it is now in the body
+  trailers.removePrefix(Envoy::Http::LowerCaseString(""));
```

Comment:
Is there a better way of clearing out the trailers?

Reply:
cc @asraa can we just add a `clear()` method to the header map? It's fine to do this as a TODO and a follow up but please add the TODO. Agree this is pretty hacky.

Reply:
Hey!! The `clear()` method is in the PR now. #9114 
Not sure which one of us will merge first, but feel free to put a TODO(asraa) in your PR and I'll follow up when mine's merged.

Reply:
Up for a master merge and cleaning this up?

Reply:
Added, thanks @asraa!